python/lsst/ts/atmcssimulator/issued_command.py

```diff
@@ -0,0 +1,114 @@
+# This file is part of ts_atmcssimulator.
+#
+# Developed for the Vera Rubin Observatory Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["IssuedCommand"]
+
+import asyncio
+
+
+class IssuedCommand:
+    """asyncio futures to track the progress of an issued low-level controller
+    command.
+
+    Parameters
+    ----------
+    timestamp : float
+        The TAI unix timestamp [sec] at which the command was issued.
+    name : str
+        The name of the command.
+
+    Attributes
+    ----------
+    ack_or_noack_timestamp : float
+        The TAI unix timestamp [sec] at which an ACK or NOACK was received.
+    success_or_fail_timestamp : float
+        The TAI unix timestamp [sec] at which a SUCCESS or FAIL was received.
+    ack : `asyncio.Future`
+        Future which ends as follows:
+
+        * result = None when the command is acknowledged.
+        * exception if the command is rejected.
+    done : `asyncio.Future`
+        Future which ends as follows:
+
+        * result = None when the command has ended successfully.
+        * exception if the command has ended in an error.
```

Comment:
Also as I point out elsewhere (and haven't gotten far enough to see what you responded) raising salobj.ExpectedError has some advantages over raising RuntimeError -- it means you can `await issued_command.done` directly in `do_x` methods in the CSC.

Reply:
As agreed during our chat today on Slack, I removed the `ack` future completely. I also removed the monitor method since it now indeed isn't necessary anymore.